### PR TITLE
e2e: sgx: fix NFD and webhook deployment

### PR DESCRIPTION
**Pull request by mythi**
Signed-off-by: Mikko Ylinen <mikko.ylinen@intel.com>

### CODE REVIEW

**Review by bart0sh — DISMISSED** [review]
/lgtm

File: test/e2e/sgx/sgx.go [hunk 1]
```diff
@@ -43,42 +41,46 @@ func init() {
...
-			framework.Failf("unable to create temp directory: %v", err)
+			framework.Failf("unable to locate %q: %v", kustomizationYaml, err)
 		}
-		defer os.RemoveAll(tmpDir)
+		_ = utils.DeployWebhook(f, kustomizationPath)
 
-		err = utils.CreateKustomizationOverlay(f.Namespace.Name, filepath.Dir(kustomizationPath), tmpDir)
-		if err != nil {
-			framework.Failf("unable to create kustomization overlay: %v", err)
+		msg := framework.RunKubectlOrDie("node-feature-discovery", "apply", "-k", "../../deployments/sgx_nfd")
```
Comment (rojkov) [inline test/e2e/sgx/sgx.go:51]: Is it safe to use a relative path to `kustomization.yaml` here given the test can be started outside of the repo's root? Probably using `utils.LocateRepoFile()` would be safer.

Comment (mythi) [reply test/e2e/sgx/sgx.go]: fixed. we're resolving the repo path quite many times. might make sense to move to Suite init at some point?

Comment (rojkov) [reply test/e2e/sgx/sgx.go]: Agree, this might be useful

**Review by rojkov — APPROVED** [review]
LGTM!

**Pull request merged**